src/mrdog/config_storage.rs

```diff
@@ -1,5 +1,93 @@
-use super::ConfigItem;
+use dirs;
+use serde::{Deserialize, Serialize};
+use std::{
+    fs, io,
+    path::{self, PathBuf},
+};
 
-pub fn set_config(item: ConfigItem) {
-    todo!("Set config {item:?}")
+use super::{ConfigValue, Provider};
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CreateFailed(io::Error),
+    CouldNotConvertToToml(toml::ser::Error),
+    WriteToConfigFailed(io::Error),
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct Config {
+    github: Option<Vec<ConfigValue>>,
+    gitlab: Option<Vec<ConfigValue>>,
+}
+
+fn update_or_push_config_value(values: &mut Vec<ConfigValue>, item: &ConfigValue) {
+    for value in values.iter_mut() {
+        if value.name == item.name {
+            value.value = item.value.clone();
+            return;
+        }
+    }
+    values.push(item.clone());
+}
+
+pub fn set_config(item: ConfigValue) -> Result<(), ConfigError> {
+    let mut config = load_or_create_config();
+
+    match item.provider {
+        Provider::Gitlab => {
+            if let Some(values) = &mut config.gitlab {
+                update_or_push_config_value(values, &item);
+            } else {
+                config.gitlab = Some(vec![item]);
+            }
+        }
+        Provider::Github => {
+            if let Some(values) = &mut config.github {
+                update_or_push_config_value(values, &item);
+            } else {
+                config.github = Some(vec![item]);
+            }
+        }
+    }
+
+    let to_set = toml::to_string(&config).map_err(ConfigError::CouldNotConvertToToml)?;
+
+    fs::write(get_config_path(), to_set.as_bytes()).map_err(ConfigError::WriteToConfigFailed)?;
+
+    Ok(())
+}
+
+fn load_or_create_config() -> Config {
+    let exists = path::Path::new(&get_config_path()).exists();
+
+    if !exists {
+        println!("Config does not exists, creating...");
+        let created_config = create_config();
+        if created_config.is_ok() {
+            println!("Created config ~/.config/mrdog.toml");
+        } else {
+            println!("Could not create a config, aborting, {created_config:?}");
+            panic!();
+        }
+    }
+
+    let contents = fs::read_to_string(get_config_path()).unwrap();
+    let tokens: Config = toml::from_str(&contents).unwrap();
+    tokens
+}
+
+fn create_config() -> Result<bool, ConfigError> {
+    let written = fs::File::create(get_config_path());
+    return written
+        .map(|_| true)
+        .map_err(|e| ConfigError::CreateFailed(e));
+}
+
+fn get_config_path() -> PathBuf {
+    let home = dirs::home_dir();
+    if let Some(p) = home {
+        p.join(".config/mrdog.toml")
```

Comment:
NIT: In my expirience, on Linux peeps tend to create a subfolder, e.g. `.config/mrdog/config.toml` to put all future trash in subfolder.

Reply:
I've checked mine, and there were both files and subfolders :D Will add a subfolder 🐱

Reply:
> I've checked mine

reminded me of this gem :D https://www.youtube.com/watch?v=c8AP4qDBfqo